File: stage1/s1_if_02.js
// S1_IF_02 — Score to grade
function grade(score) {
  if (typeof score !== "number" || score < 0 || score > 100) {
    return null;
  }

  if (score >= 90) return "A";
  if (score >= 80) return "B";
  if (score >= 70) return "C";
  if (score >= 60) return "D";
  return "F";
}

// Tests
console.log(grade(95)); // "A"
console.log(grade(82)); // "B"
console.log(grade(74)); // "C"
console.log(grade(61)); // "D"
console.log(grade(40)); // "F"
console.log(grade(-5)); // null
console.log(grade(120)); // null

File: stage1/B/S1_IF_02.py
# S1_IF_02.py

def grade(score):
    if not isinstance(score, (int, float)) or isinstance(score, bool):
        return None
    if score < 0 or score > 100:
        return None

    if score >= 90:
        return "A"
    elif score >= 80:
        return "B"
    elif score >= 70:
        return "C"
    elif score >= 60:
        return "D"
    else:
        return "F"


def main():
    tests = [59, 60, 69, 70, 89, 90, 100, 101]
    for s in tests:
        print(f"score={s:<3} -> {grade(s)!r}")


if __name__ == "__main__":
    main()
